src/compiler/factory/nodeFactory.ts

```diff
@@ -7570,6 +7583,9 @@ export function createSourceMapSource(fileName: string, text: string, skipTrivia
 
 // Utilities
 
+/** @internal */ export function setOriginalNode<T extends Node>(node: Mutable<T>, original: Node | undefined): T; // TODO: This should probably always require a Mutable<T>, since it mutates `original`
+// eslint-disable-next-line @typescript-eslint/unified-signatures
+export function setOriginalNode<T extends Node>(node: T, original: Node | undefined): T; // inference is worse if the signatures are combined

```

Comment:
Does this comment show up in the public API? It would be somewhat confusing if so, since the other overload isn't public.